### PR TITLE
Only change order line status when status is available

### PR DESCRIPTION
**Pull request by viggo-devries**
It's possible for lines to have a completely different status than the order.
When a line is at an "end status" where it cannot change anymore this will result in the line being changed to something you might not want.

### CODE REVIEW

**Review by specialunderwear — CHANGES_REQUESTED** [review]
please add a test

**Comment by sasha0** [comment]
Needs test.

**Comment by viggo-devries** [comment]
hahah definitely tests, was curious if the current tests would still pass.

File: src/oscar/apps/order/abstract_models.py [hunk 1]
```diff
@@ -135,8 +135,9 @@ def set_status(self, new_status):
         self.status = new_status
         if new_status in self.cascade:
             for line in self.lines.all():
-                line.status = self.cascade[self.status]
-                line.save()
+                if new_status in line.available_statuses():
```
Comment (specialunderwear) [inline src/oscar/apps/order/abstract_models.py:138]: ```if self.cascade[new_status] in line.available_statuses():```

**Review by specialunderwear — CHANGES_REQUESTED** [review]

**Review by joeyjurjens — APPROVED** [review]

**Review by specialunderwear — APPROVED** [review]

**Pull request merged**